Remove UI screenshots

emlys opened the pull request:
Fixes #15 
- Remove all screenshots of UI
- Remove unused files and add a step to the actions workflow that checks for unused files
- Replace sample tables with tables from the invest-sample-data repo
- Added `make sampledata` command that checks out the sample data
    - For a few sample data tables, makes a modified version (shorten long tables, replace \ with \\)

emlys commented on source/reservoirhydropowerproduction.rst:
```diff
@@ -389,31 +389,6 @@ The simplest categorization of LULCs on the landscape involves delineation by la
-  17     Orchards/Vineyards
-  18     Pasture
-  ====== ===========================
-
```
what do you think about these? the same LULC section with table is repeated in a few models. I took this one out because it seemed unnecessary and potentially confusing (this table is not actually an input to anything)... but maybe it is useful to explain the concept of landcover codes.

dcdenu4 replied:
Yea, I ultimately think it's useful in this context since this is an *appendix* zone calling out useful data sources nuggets. 

However, I wish that we had a User's Guide section called **InVEST Data Resources** (or something), where we collated information on inputs for ALL models that we could point users to and update as we learned more (like from forum conversations).

emlys replied:
Yea that would be great! Made an issue for it: #23

dcdenu4 commented on .github/workflows/build-sphinx.yml:
```diff
@@ -32,6 +32,30 @@ jobs:
             - name: Install python dependencies
               run: python -m pip install -r requirements.txt
 
+            - name: Check for unused files
```
I tried thinking of a situation where we wouldn't mind having an unused file hanging around and I couldn't really come up with anything. I think it's fine to be strict and clean about this until a situation arises that makes us think differently.

emlys replied:
Agree, they were really piling up and I was using this script to find files to delete anyway, so I figured I'd add it to the workflow

dcdenu4 commented on Makefile:
```diff
@@ -33,3 +33,33 @@ linkcheck: $(SOURCEDIR)
+	sed 's/\\/\\\\/g' invest-sample-data/HabitatRiskAssess/Input/habitat_stressor_info.csv > invest-sample-data/HabitatRiskAssess/Input/habitat_stressor_info_modified.csv
+
+	# selections of tables that are too long to display in full
+	head -n1 invest-sample-data/pollination/landcover_biophysical_table.csv > invest-sample-data/pollination/landcover_biophysical_table_modified.csv 
+	tail -n3 invest-sample-data/pollination/landcover_biophysical_table.csv >> invest-sample-data/pollination/landcover_biophysical_table_modified.csv 
```
nifty!

dcdenu4 commented on Makefile:
```diff
@@ -33,3 +33,33 @@ linkcheck: $(SOURCEDIR)
+
+	head -n4 invest-sample-data/WindEnergy/input/NE_sub_pts.csv > invest-sample-data/WindEnergy/input/NE_sub_pts_modified.csv
+
+
+
+
+	
```
Some dead space here.

dcdenu4 commented on source/carbon_edge.rst:
```diff
@@ -196,34 +196,18 @@ The biophysical table columns must be named as follows:
+          :header-rows: 1
+          :widths: auto
+
+   *Sources for c_above data*: 
+
+   Closed shrublands, open shrublands, grasslands: Miranda, Sabrina do Couto, et al. Regional variations in biomass distribution in Brazilian savanna woodland. Biotropica 46.2 (2014): 125-138. 
+
+   Woody savannas, savanna: Saatchi, Susan S., et al. Distribution of aboveground live biomass in the Amazon basin. Global Change Biology 13.4 (2007): 816-837. 
+
+   Croplands: Fearnside, Philip M. Greenhouse gases from deforestation in Brazilian Amazonia: net committed emissions. Climatic Change 35.3 (1997): 321-360.
```
I notice that we were super-scripting values in the table to indicate which source they came from... This is probably information we should try to replicate if we can. Maybe there's some clean way to say which rows from the table go with which source for the `c_above` values?

dcdenu4 replied:
I also noticed this table gets a `Table 4` designation, but it's the first table in this doc...

emlys replied:
I was sort of trying to replicate it by putting `*Sources for c_above data*: ` and then the row descriptions but it's not as nice as the superscripts. Side note, it would be great to have this source information for all sample data! I'm not always sure if they're real data or just made up.

dcdenu4 commented on source/carbonstorage.rst:
```diff
@@ -87,17 +87,10 @@ This section outlines the specific data used by the model. See the Appendix for
-  4      Shrub/undergrowth   30      30      30     13
-  5      Open/urban          5       5       15     2
-  ====== ================== ======= ======= ====== ======
+  .. csv-table:: **Example Carbon Pools Table**
+       :file: ../invest-sample-data/Carbon/carbon_pools_willamette_modified.csv
+       :header-rows: 1
+       :widths: auto
```
I wonder if we can still include the table description here... not that it really is providing much context you couldn't get from looking at the table values.

dcdenu4 replied:
Like the carbon edge table, this one is labeled `Table 5`. I guess it's just incrementing on the whole doc...

emlys replied:
I put the description back and changed it to make sense with the data

dcdenu4 commented on source/coastal_vulnerability.rst:
```diff
@@ -61,7 +61,7 @@ Table 4.1
 
 .. csv-table:: **Example Ranking System**
-      :file: ./coastal_vulnerability/cv_table_41.csv
+      :file: ./coastal_vulnerability/ranking_system_table.csv
```
This looks great. I am seeing `Table 8` instead of the before `Table 4`. The `Table 4.1: List ...` then doesn't line up... Not sure what's throwing off table numbers.

dcdenu4 commented on source/coastal_vulnerability.rst:
```diff
@@ -340,8 +340,9 @@ The model uses a graphical interface to input all required and optional data. Be
 
    .. csv-table:: **Example Habitat Table**
-      :file: ./coastal_vulnerability/NaturalHabitat_WCVI.csv
+      :file: ../invest-sample-data/CoastalVulnerability/GrandBahama_Habitats/Natural_Habitats.csv
```
Again just mentioning table number differences.

dcdenu4 commented on source/croppollination.rst:
```diff
@@ -159,13 +159,16 @@ This section outlines the specific data used by the model. See the Appendix for
 
 .. csv-table:: **Example Biophysical Table**
-       :file: ./croppollination/landcover_biophysical_table_sample.csv
+       :file: ../invest-sample-data/pollination/landcover_biophysical_table_modified.csv
```
This table has `Table 2` instead of original `Table 1`. It also has a column `descript`. Should that really be `description` and changed in the sample data? I'm assuming the model doesn't care and never looks for such a column, but it does look a little sloppy in the Users' Guide.

emlys replied:
Hmm yea. May as well change that in the sample data.

dcdenu4 commented on source/croppollination.rst:
```diff
@@ -187,11 +190,9 @@ This section outlines the specific data used by the model. See the Appendix for
 
 .. csv-table:: **Example Guild Table**
-       :file: ./croppollination/guild_table_sample.csv
+       :file: ../invest-sample-data/pollination/guild_table.csv
```
This is showing as `Table 3` instead of original `Table 2`.

dcdenu4 commented on source/globio.rst:
```diff
@@ -513,37 +499,10 @@ lucode globio_lucode
-msa_lu        Land Cover Class 9           0.1   0.08
-msa_lu        Land Cover Class 10          0.05  na
-============= ================ =========== ===== =====
+    .. csv-table:: **Example**
```
This table listed as `Table 10` as opposed to be unlisted originally.

dcdenu4 commented on source/habitat_risk_assessment.rst:
```diff
@@ -425,9 +423,13 @@ Criteria Scores CSV
 The Criteria Scores CSV (or Excel) file will provide all the criteria information for the run of the Habitat and Species Risk Assessment model. This file contains information about the effect of each stressor on each habitat (i.e. the exposure and consequence scores) for the habitats and stressors in your analysis. A template for the criteria CSV file can be found in the sample data folder. Users ...
 
-.. figure:: habitat_risk_assessment/criteria_csv.PNG
+.. This image is of the same table provided in sample data.
+   It was too large and complex to format well as a csv-table so I'm leaving it as a figure.
```
Yup, makes sense.

dcdenu4 commented on source/ndr.rst:
```diff
@@ -238,9 +238,10 @@ You may choose to run the model with either Nitrogen or Phosphorus or both at th
 
   .. csv-table::
-    :file: ndr/ndr_biophysical_table_example.csv
+    :file: ../invest-sample-data/NDR/biophysical_table_gura.csv
```
Do you think that showing a table with extra columns in this case is a little confusing? I also notice that the sample data uses `eff_p` as opposed to `eff_n`. I feel like in this case the sample data might not be robust to helping explain the capabilities of the model? 

I also noticed that some of the descriptions above mention things like:
>For example, high values (0.6 to 0.8) may be assigned to all natural vegetation types (such as forests, natural pastures, wetlands, or prairie)

These don't necessarily jive (jibe?) with what we have in the sample data.

dcdenu4 commented on source/recreation.rst:
```diff
@@ -114,8 +114,9 @@ The following outlines the options presented to the user after opening the InVES
 
     .. csv-table:: **Example Predictor Table**
-       :file: ./recreation/predictors.csv
+       :file: ../invest-sample-data/recreation/predictors.csv
```
Just noting that the table is `Table 6` as opposed to the original `Table 3`.

dcdenu4 commented on source/sdr.rst:
```diff
@@ -471,32 +469,6 @@ The simplest categorization of LULCs on the landscape involves delineation by la
 
 A slightly more sophisticated LULC classification involves breaking relevant LULC types into more meaningful types. For example, agricultural land classes could be broken up into different crop types or forest could be broken up into specific species. The categorization of land use types depends on the model and how much data is available for each of the land types. You should only break up a lan ...
 
-*Sample Land Use/Land Cover Table*
-
-  ====== ===========================
-  lucode Land Use/Land Cover
-  ====== ===========================
-  1      Evergreen Needleleaf Forest
```
Ah, interesting. This is annoyingly redundant. I wonder, outside of it's own section like I mentioned above, if this is a case where having a CSV stored in the UG repo makes sense, so it can be re-used in places like this? 

Unless we decide this isn't helpful at all.

dcdenu4 commented on source/wave_energy.rst:
```diff
@@ -80,11 +78,11 @@ The ability of a WEC device to harvest wave energy can be expressed by wave ener
-.. figure:: ./wave_energy/table_energyabsorption.png
-   :align: center
-   :figwidth: 500px
+.. csv-table:: **Wave energy absorption performance (kW) for Pelamis**
```
A few things here:
- I noticed the `Figure` number changed. I also noticed a reference to `Table 3.1` that might have been that commented out `image002`? 
- I wonder if keeping this `png` table makes sense here, but later on for the **Data Needs**, it makes sense to show what that table really looks like as a data input?

emlys replied:
The `Table 3.1` reference seems to be about the seastate table. I just change it to say "The table below". And I'll put the png back here.

dcdenu4 commented on source/wave_energy.rst:
```diff
@@ -148,19 +146,21 @@ First we describe required inputs.  The required inputs are the minimum data nee
-.. figure:: ./wave_energy/table_pelamisperformance.png
-   :align: center
-   :figwidth: 500px
+.. csv-table::
```
The indentation between the numbered inputs and the associated tables doesn't look great, is it possible to bump out the tables too? I also wonder if using your modifying approach might be good here to truncate some of these larger tables where more rows / cols doesn't really provide a lot of benefit?

emlys replied:
I had tried to line up the tables within the numbered items and had no luck but someone on stackoverflow just pointed me in the right direction so it's fixed now!

dcdenu4 reviewed (changes requested):
Hey @emlys,

This is an interesting challenge! Overall I like the idea that UG model examples should be based on the sample data and having one source of truth for that makes sense. However, I go back and forth on how to think of the UG. Is it more of a publication or more of a detailed how-to guide? If the model developers think it should read more like a publication then maybe some of those tables / figures that don't really represent the sample data, but do provide a more simple example of understanding make sense. In some cases where the UG already is trying to reflect the sample data, then I think it makes total sense to replace those with the direct sample data connection. 

Some other things I noticed that should be looked at more closely:
- Table and figure numbers have changed significantly. I don't think a lot of the models reference these numbers, but some of them do and no longer reads well.
- Do we need to make changes to how InVEST `make userguide` operates as well?
- I'm assuming all those `crop_production` files that were deleted were unused? Are those files represented in the sample data? I'm just wondering if those would / were useful to someone and they should live elsewhere.
- I noticed that the `Fisheries` table now had some vertical table lines that make an empty column

Happy to chat about this more!

dcdenu4 commented:
Hey @emlys , is this PR back to me? No rush, I just wanted to make sure I wasn't holding anything up. Happy to chat about this more as well!

emlys commented:
@dcdenu4 Sorry, I've been totally focused on the stormwater model! I addressed a bunch of the table issues but I don't think I quite got to all of your questions. Is it alright to leave this for a couple more days?

emlys commented:
> If the model developers think it should read more like a publication then maybe some of those tables / figures that don't really represent the sample data, but do provide a more simple example of understanding make sense. In some cases where the UG already is trying to reflect the sample data, then I think it makes total sense to replace those with the direct sample data connection.

That's a good point, after going back thru it I agree some make more sense as a simplified example that's specific to the user's guide. I put back the original tables in some cases that you pointed out above. Let me know what you think

> Table and figure numbers have changed significantly. I don't think a lot of the models reference these numbers, but some of them do and no longer reads well.

Hmm, yea. Realizing I really don't know how the table numbering works. After a bit of googling I think it's a problem to put off until really necessary. Sphinx wants to number them relative to the entire user's guide, not by page. For now I've just removed the table headers that trigger auto-numbering. We can reference them by the `:label:` instead.

> Do we need to make changes to how InVEST `make userguide` operates as well?

I think it's unaffected

> I'm assuming all those `crop_production` files that were deleted were unused? Are those files represented in the sample data? I'm just wondering if those would / were useful to someone and they should live elsewhere.

There was some overlap with the sample data but not entirely. I guess if someone needs them in the future, they'll be there in the git history?

> I noticed that the `Fisheries` table now had some vertical table lines that make an empty column

I added the vertical lines to all the tables since some were hard to read without it. That empty column is in the sample data and it does look bad. I'll remove it from that table.

emlys commented:
Also addressing https://github.com/natcap/invest.users-guide/issues/22 by
- configuring the user-agent
- setting `linkcheck` to ignore a troublesome link
- adding a 2 minute timeout to the actions step to prevent the issue where it ran for 6 hours

dcdenu4 commented on source/coastal_vulnerability.rst:
```diff
@@ -53,18 +53,12 @@ The primary output of the model is a geospatial dataset (points) that are plotte
-
-.. csv-table:: **Example Ranking System**
-      :file: ./coastal_vulnerability/cv_table_41.csv
+.. csv-table:: **Example Ranking Table**
```
Just a nit-pick here, but can we get a little space between the table and 
>The model calculates the exposure...

dcdenu4 commented on source/wave_energy.rst:
```diff
@@ -59,32 +59,25 @@ We prepared globally and regionally available input data layers for the potentia
    :align: center
    :figwidth: 500px
 
-   Wave energy absorption performance (kW) in each seastate bin for Pelamis.
```
Did we want to keep this in?

dcdenu4 reviewed (changes requested):
Hey @emlys ,

Thanks for making another pass at this and fixing some of the table numbering issues. I just a few small comments going through this again. But it's looking pretty much good to go.

dcdenu4 reviewed (approved):
Alright @emlys let's make this canonical!
